### PR TITLE
Add BYOI functionality to Connectors CLI

### PR DESCRIPTION
**Pull request by navarone-feekery**
**Closes https://github.com/elastic/enterprise-search-team/issues/6255**
**Closes https://github.com/elastic/enterprise-search-team/issues/6273**

---

- Allow for the Connectors CLI `create` function to create connectors for indices that already exist.
- Remove `search-` prefix requirements
  - The `search-` prefix will be forcibly added to a connector created in CLI if it's intended to be native
- Allow rudimentary connector client creation; the API key management still needs to be completed in Kibana (we will solve this issue later).
- Add checks for existing index/connector and error out appropriately
- Include index name in final success output

New commands:
```bash
# To create a connector client
$ ./bin/connectors connector create

# To create a native connector
$ ./bin/connectors connector create --native

# To use BYOI
$ ./bin/connectors connector create --from_index
```

Help outputs:

```bash
$ ./bin/connectors connector --help
Usage: connectors connector [OPTIONS] COMMAND [ARGS]...

  Connectors management

Options:
  --help  Show this message and exit.

Commands:
  create  Creates a new connector and a search index
  list    List all existing connectors


$ ./bin/connectors connector create --help
Usage: connectors connector create [OPTIONS]

  Creates a new connector and a search index

Options:
  --index_name TEXT               Name of the index. If the index is native,
                                  `search-` will be prepended to the input.
  --service_type [azure_blob_storage|confluence|dir|dropbox|github|gmail|google_cloud_storage|google_drive|jira|mongodb|mssql|mysql|network_drive|onedrive|oracle|outlook|postgresql|s3|salesforce|servicenow|sharepoint_online|sharepoint_server|slack|microsoft_teams|zoom|box]
  --index_language TEXT
  --native                        Create a native connector rather than a
                                  connector client.
  --from_index                    Create a connector for an index that already
                                  exists. Each index can only have one
                                  connector. All current docs for the index
                                  will be deleted during the first sync.
  --help                          Show this message and exit.
```

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [x] Considered corresponding documentation changes
- [x] Contributed any configuration settings changes to the configuration reference

## Release Note

You can use the Connectors CLI to also convert an existing index into a connector, by including the flag `--from_index`. The target index cannot already be a connector. This works for both native connectors and connector clients.

### CODE REVIEW

**Comment by vidok** [comment]
Could you please add `./bin/connectors connector --help` output to the PR description?

File: connectors/connectors_cli.py [hunk 1]
```diff
@@ -149,7 +149,8 @@ def validate_language(ctx, param, value):
 @click.command(help="Creates a new connector and a search index")
 @click.option(
     "--index_name",
-    prompt=f"{click.style('?', blink=True, fg='green')} Search index name (search-)",
+    prompt=f"{click.style('?', blink=True, fg='green')} Index name",
+    help="Name of the index. If the index is native, `search-` will be prepended to the input.",
```
[the remark this answers is not in the document]
Comment (navarone-feekery) [reply connectors/connectors_cli.py:153]: I meant if the Connector is intended to be native. I'll reword it.

File: connectors/connectors_cli.py [hunk 2]
```diff
@@ -177,6 +199,35 @@ def prompt():
...
+
+    if index_exists and not use_existing_index:
+        click.echo(
+            click.style(
+                f"Index for {index_name} already exists. Include the flag `--use_existing_index` to create a connector for this index.",
+                fg="red",
+            ),
+            err=True,
+        )
+        return
```
Comment (vidok) [inline connectors/connectors_cli.py:212]: What is the status of the command in this case?

Comment (navarone-feekery) [reply connectors/connectors_cli.py:212]: Should I raise the error instead? I didn't think to ensure the exit code isn't `0`.
Or maybe this is better: https://click.palletsprojects.com/en/8.0.x/api/#click.Abort

Comment (navarone-feekery) [reply connectors/connectors_cli.py]: Added `raise click.Abort()` to the index/connector exists catches

File: connectors/cli/connector.py [hunk 3]
```diff
@@ -43,25 +43,49 @@ async def list_connectors(self):
 
     def service_type_configuration(self, source_class):
         source_klass = get_source_klass(source_class)
-        configuration = source_klass.get_default_configuration()
+        configuration = source_klass.get_simple_configuration()
```
[the remark this answers is not in the document]
Comment (seanstory) [reply connectors/cli/connector.py:46]: we should rename `get_simple_configuration()` as it's actually the more complex/detailed of the two. :/

**Review by vidok — APPROVED** [review]
Great job 🔥

File: connectors/cli/connector.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (navarone-feekery) [reply connectors/cli/connector.py:46]: @seanstory I'll follow this up in another PR, but I agree the naming is misleading.

**Pull request merged**